src/algebra/star/module.lean

```diff
@@ -33,3 +33,35 @@ def star_linear_equiv (R : Type*) {A : Type*}
 { to_fun := star,
   map_smul' := star_smul,
   .. star_add_equiv }
+
+variables (R : Type*) {A : Type*}
+  [semiring R] [invertible (2 : R)] [star_monoid R] [has_trivial_star R]
+  [add_comm_group A] [module R A] [star_add_monoid A] [star_module R A]
+
+/-- The self-adjoint part of an element of a star module, as a linear map. -/
+@[simps] def self_adjoint_part : A →ₗ[R] self_adjoint A :=
```

Comment:
I'm not sure about the naming here but I prefer `comp` to `part`.
```suggestion
@[simps] def self_adjoint_comp : A →ₗ[R] self_adjoint A :=
```

Reply:
I'm not sure about this -- to me `comp` really sounds like "composition" when it comes to mathlib. Maybe @eric-wieser could weigh in?

Reply:
I agree, `comp` is worse than `part`. We already use `part` for Jordan decompositions, so I think this is fine